pkg/kubehound/core/core.go

```diff
@@ -3,271 +3,145 @@ package core
-
-	// Setup telemetry
-	log.I.Info("Initializing application telemetry")
-	ts, err := telemetry.Initialize(cfg)
-	if err != nil {
-		log.I.Warnf("failed telemetry initialization: %v", err)
-	}
-	defer telemetry.Shutdown(ts)
-
+// Initiating all the providers need for KubeHound (cache, store, graph)
+func NewProvidersFactoryConfig(ctx context.Context, lc *LaunchConfig) (*ProvidersFactoryConfig, error) {
 	// Create the cache client
 	log.I.Info("Loading cache provider")
-	cp, err := cache.Factory(ctx, cfg)
+	cp, err := cache.Factory(ctx, lc.Cfg)
 	if err != nil {
-		return fmt.Errorf("cache client creation: %w", err)
+		return nil, fmt.Errorf("cache client creation: %w", err)
 	}
-	defer cp.Close(ctx)
 	log.I.Infof("Loaded %s cache provider", cp.Name())
 
 	// Create the store client
 	log.I.Info("Loading store database provider")
-	sp, err := storedb.Factory(ctx, cfg)
+	sp, err := storedb.Factory(ctx, lc.Cfg)
 	if err != nil {
-		return fmt.Errorf("store database client creation: %w", err)
+		return nil, fmt.Errorf("store database client creation: %w", err)
 	}
-	defer sp.Close(ctx)
 	log.I.Infof("Loaded %s store provider", sp.Name())
 
 	err = sp.Prepare(ctx)
 	if err != nil {
-		return fmt.Errorf("store database prepare: %w", err)
+		return nil, fmt.Errorf("store database prepare: %w", err)
 	}
 
 	// Create the graph client
 	log.I.Info("Loading graph database provider")
-	gp, err := graphdb.Factory(ctx, cfg)
+	gp, err := graphdb.Factory(ctx, lc.Cfg)
 	if err != nil {
-		return fmt.Errorf("graph database client creation: %w", err)
+		return nil, fmt.Errorf("graph database client creation: %w", err)
 	}
-	defer gp.Close(ctx)
 	log.I.Infof("Loaded %s graph provider", gp.Name())
 
 	err = gp.Prepare(ctx)
 	if err != nil {
-		return fmt.Errorf("graph database prepare: %w", err)
+		return nil, fmt.Errorf("graph database prepare: %w", err)
 	}
 
-	// Create the collector instance
-	log.I.Info("Loading Kubernetes data collector client")
-	collect, err := collector.ClientFactory(ctx, cfg)
-	if err != nil {
-		return fmt.Errorf("collector client creation: %w", err)
-	}
-	defer collect.Close(ctx)
-	log.I.Infof("Loaded %s collector client", collect.Name())
+	return &ProvidersFactoryConfig{
+		CacheProvider: cp,
+		StoreProvider: sp,
+		GraphProvider: gp,
+	}, nil
+}
 
-	// All dependencies are loaded - we can complete the config with runtime information
-	cluster, err := collect.ClusterInfo(ctx)
-	if err != nil {
-		return fmt.Errorf("collector cluster info: %w", err)
-	}
-	cfg.ComputeDynamic(config.WithClusterName(cluster.Name))
+func (fc *ProvidersFactoryConfig) Close(ctx context.Context) {
+	fc.CacheProvider.Close(ctx)
+	fc.StoreProvider.Close(ctx)
+	fc.GraphProvider.Close(ctx)
+}
 
-	// Run the ingest pipeline
-	log.I.Info("Starting Kubernetes raw data ingest")
-	if err := ingestData(ctx, cfg, collect, cp, sp, gp); err != nil {
-		return fmt.Errorf("raw data ingest: %w", err)
-	}
+type LaunchConfig struct {
+	Cfg      *config.KubehoundConfig
```

Comment:
Do we want to expose Cfg to other package?
I don't think it's used anywhere outside this pacakge by looking at the code locally.

Reply:
yes it is being used, you will see in the next PRs